servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SslCertificateCompressionTest.java

```diff
@@ -0,0 +1,179 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+import io.servicetalk.client.api.TransportObserverConnectionFactoryFilter;
+import io.servicetalk.http.api.BlockingHttpClient;
+import io.servicetalk.http.api.HttpServerBuilder;
+import io.servicetalk.http.api.HttpService;
+import io.servicetalk.http.api.SingleAddressHttpClientBuilder;
+import io.servicetalk.test.resources.DefaultTestCerts;
+import io.servicetalk.transport.api.CertificateCompressionAlgorithms;
+import io.servicetalk.transport.api.ClientSslConfigBuilder;
+import io.servicetalk.transport.api.ConnectionInfo;
+import io.servicetalk.transport.api.ConnectionObserver;
+import io.servicetalk.transport.api.HostAndPort;
+import io.servicetalk.transport.api.ServerContext;
+import io.servicetalk.transport.api.ServerSslConfigBuilder;
+import io.servicetalk.transport.api.SslProvider;
+import io.servicetalk.transport.api.TransportObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver;
+
+import org.junit.jupiter.api.Test;
+
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLSession;
+
+import static io.servicetalk.concurrent.api.Single.succeeded;
+import static io.servicetalk.http.api.HttpSerializers.textSerializerUtf8;
+import static io.servicetalk.test.resources.DefaultTestCerts.serverPemHostname;
+import static io.servicetalk.transport.netty.internal.AddressUtils.localAddress;
+import static io.servicetalk.transport.netty.internal.AddressUtils.serverHostAndPort;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+class SslCertificateCompressionTest {
+
+    /**
+     * Compares the bytes written and read when certificate compression is enabled vs. when it is not.
+     */
+    @Test
+    void negotiatesServerCertCompression() throws Exception {
+        final HttpService service = (ctx, request, responseFactory) ->
+            succeeded(responseFactory.ok().payloadBody("Hello World!", textSerializerUtf8()));
+
+        long readWithoutCompression;
+        long readWithCompression;
+        long writtenWithoutCompression;
+        long writtenWithCompression;
+
+        try (ServerContext server = serverBuilder(false).listenAndAwait(service)) {
+            SslBytesReadTransportObserver observer = new SslBytesReadTransportObserver();
+            try (BlockingHttpClient client = clientBuilder(server, false, observer).buildBlocking()) {
```

Comment:
Even if the client-side compression currently doesn't work in netty, consider making this test parametrized and run it with/without mTLS to cover both cases.